osf_tests/test_registration_moderation_notifications.py

```diff
@@ -512,8 +533,28 @@ def test_submissions_and_withdrawals_both_appear_in_moderator_digest(self, diges
 
     @pytest.mark.parametrize('digest_type', ['email_transactional', 'email_digest'])
     def test_submsissions_and_withdrawals_do_not_appear_in_node_digest(self, digest_type, registration, admin, moderator, daily_moderator):
-        with mock.patch('website.reviews.listeners.mails.send_mail'):
-            notify_submit(registration, admin)
+        notify_submit(registration, admin)
         notify_moderator_registration_requests_withdrawal(registration, admin)
 
-        assert not list(get_users_emails(digest_type))
+        assert not list(tasks.get_users_emails(digest_type))
+
+    def test_moderator_digest_emails_render(self, registration, admin, moderator):
+        notify_moderator_registration_requests_withdrawal(registration, admin)
+        # Set up mock_send_mail as a pass-through to the original function.
+        # This lets us assert on the call count/args and also implicitly
+        # ensures that the email acutally renders as normal in send_mail.
+        send_mail = mails.send_mail
+        with mock.patch.object(tasks.mails, 'send_mail', side_effect=send_mail) as mock_send_mail:
+            tasks._send_reviews_moderator_emails('email_transactional')
+
+        mock_send_mail.assert_called()
+
+    def test_branded_provider_notification_renders(self, registration, admin, moderator):
+        provider = registration.provider
+        provider.brand = Brand.objects.create(hero_logo_image='not-a-url', primary_color='#FFA500')
+        provider.name = 'Test Provider'
+        provider.save()
+
+        notify_submit(registration, admin)
+        tasks._send_reviews_moderator_emails('email_transactional')
+        assert True  # everything rendered!
```

Comment:
I don't understand why this is necessary.

Reply:
`notify_submit` calls send_mail using a template that inherits from `notify_base.mako` and populates it with context that's set in `utils/notifications.py`, it also populates entires for the digest email.

`_send_Reviews_moderator_emails` populates the digest email, which inherits from `notify_base.mako` and populates it with context that's set in `website/notifications/tasks.py` (but only if there are actual messages for the reviews email)

We want to make sure that the notify_base template renders successfully in both of these places when we're working with a branded registration provider. Will add documentation to that effect.

The final assert is meaningless and just demonstrates that we didn't raise an exception prior due to failed rendering